app/src/main/java/com/douglasbruce/tasky/features/event/EventViewModel.kt

```diff
@@ -3,25 +3,41 @@ package com.douglasbruce.tasky.features.event
 import androidx.compose.runtime.mutableStateOf
 import androidx.lifecycle.SavedStateHandle
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
 import androidx.lifecycle.viewmodel.compose.SavedStateHandleSaveableApi
 import androidx.lifecycle.viewmodel.compose.saveable
+import com.douglasbruce.tasky.core.common.auth.AuthResult
 import com.douglasbruce.tasky.core.common.utils.DateUtils
+import com.douglasbruce.tasky.core.domain.datastore.UserDataPreferences
+import com.douglasbruce.tasky.core.domain.repository.EventRepository
+import com.douglasbruce.tasky.core.model.AgendaItem
 import com.douglasbruce.tasky.core.model.AgendaPhoto
+import com.douglasbruce.tasky.core.model.NotificationType
 import com.douglasbruce.tasky.features.event.form.EventFormEvent
 import com.douglasbruce.tasky.features.event.form.EventState
 import com.douglasbruce.tasky.features.event.navigation.EventArgs
 import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
 import java.time.LocalTime
+import java.time.ZonedDateTime
 import java.time.temporal.ChronoUnit
+import java.util.UUID
 import javax.inject.Inject
 
 @OptIn(SavedStateHandleSaveableApi::class)
 @HiltViewModel
 class EventViewModel @Inject constructor(
     savedStateHandle: SavedStateHandle,
+    private val eventRepository: EventRepository,
+    private val userDataPreferences: UserDataPreferences,
 ) : ViewModel() {
 
     private val eventArgs: EventArgs = EventArgs(savedStateHandle)
+    private val localUserId: String
+        get() = runBlocking { userDataPreferences.userData.map { it.userId }.first() }
```

Comment:
This is not ideal, since it blocks the main thread every time you read the user ID (as long as the coroutine is running on `Dispatchers.Main` at least). You could rather just expose the flow and then call `first()` when you need it:
```
private val localUserId: Flow<String> = userDataPreferences.userData.map { it.userId }

// ...

viewModelScope.launch {
    val userId = localUserId.first()
}
```